### PR TITLE
[Redesign] Fixing the amount input and when chains/tokens are swapped and handling couple of error cases

### PR DESCRIPTION
**Pull request by emreboga**
- We need to clear the amount input when chains/tokens are swapped.
- We also need to add a couple of checks for error cases.

### CODE REVIEW

File: wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx [hunk 1]
```diff
@@ -66,7 +66,6 @@ const AmountInput = (props: Props) => {
     route,
   } = useSelector((state: RootState) => state.transferInput);
 
-  const [tokenAmount, setTokenAmount] = useState(amount);
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx:69]: I've decided to remove the local state var and directly use amount in Redux . That's why I also moved the validation in useEffect into the amount change handler.

File: wormhole-connect/src/views/v2/Bridge/SwapInputs/index.tsx [hunk 2]
```diff
@@ -33,12 +33,13 @@ function SwapInputs() {
     fromChain &&
     !config.chains[fromChain]?.disabledAsDestination &&
     toChain &&
-    !config.chains[toChain]?.disabledAsDestination;
+    !config.chains[toChain]?.disabledAsSource;
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/SwapInputs/index.tsx:36]: I think this was a copy/paste error, or am I missing anything here?

File: wormhole-connect/src/views/v2/Bridge/Routes/SingleRoute.tsx [hunk 3]
```diff
@@ -224,6 +224,10 @@ const SingleRoute = (props: Props) => {
   }, [props.route.name]);
 
   const providerText = useMemo(() => {
+    if (!sourceToken) {
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/Routes/SingleRoute.tsx:227]: This is a very rare case (I've seen it when switching chains), but good to be safe.

File: wormhole-connect/src/views/v2/Bridge/SwapInputs/index.tsx (hunk 2 above)
Comment (artursapek) [reply wormhole-connect/src/views/v2/Bridge/SwapInputs/index.tsx:36]: came from here https://github.com/wormhole-foundation/wormhole-connect/commit/5c13b8d1aea40d1ee896deb4c0e7a375cffdbb02#diff-50062b7f121564dcafd21f3ee9350314aae20a3a4b2487eff668042c0d358587R36

**Review by artursapek — APPROVED** [review]

File: wormhole-connect/src/views/v2/Bridge/Routes/SingleRoute.tsx (hunk 3 above)
Comment (artursapek) [reply wormhole-connect/src/views/v2/Bridge/Routes/SingleRoute.tsx:227]: came from here https://github.com/wormhole-foundation/wormhole-connect/commit/5c13b8d1aea40d1ee896deb4c0e7a375cffdbb02#diff-50062b7f121564dcafd21f3ee9350314aae20a3a4b2487eff668042c0d358587R36

**Pull request merged**